platform-sdk/swirlds-benchmarks/build.gradle.kts

```diff
@@ -50,10 +50,22 @@ fun listProperty(value: String) = objects.listProperty<String>().value(listOf(va
 
 tasks.register<JMHTask>("jmhReconnect") {
     includes.set(listOf("Reconnect.*"))
-    jvmArgs.set(listOf("-Xmx16g"))
+    jvmArgs.set(
+        listOf(
+            "-Xmx16g",
+            "-Xms16g",
+            "-XX:+UnlockExperimentalVMOptions",
+            "-XX:+UseZGC",
+            "-XX:MaxDirectMemorySize=48g"
+        )
+    )
 
     resultsFile.convention(layout.buildDirectory.file("results/jmh/results-reconnect.txt"))
 
-    benchmarkParameters.put("numRecords", listProperty("100000"))
-    benchmarkParameters.put("numFiles", listProperty("500"))
+    benchmarkParameters.put("numRecords", listProperty("1000"))
+    benchmarkParameters.put("numFiles", listProperty("10"))
+    benchmarkParameters.put("delayMilliseconds", listProperty("1"))
```

Comment:
Would it be appropriate to fuzz this value?
Real networks have variable latency (and typically range from 2ms to 500ms or more, depending on route).  Perhaps config for minDelayMilliseconds and maxDelayMilliseconds then the slow reconnect implementations insert a value that's random between min and max?
Just a thought that the variability may help us find more potential issues.

Reply:
I was thinking about that myself. Two reasons I decided against this at this time:
1. I already have to use a 1 ms delay. Anything larger would make the benchmark run a lot longer because the delay applies to every message exchanged over the wire. While one could use nanoTime() and a busy sleep loop for delays less than 1 ms, I don't find this elegant. Especially when it comes to performance testing.
2. It's nice when benchmark results are as reproducible as possible. Introducing randomness into the delays, especially when benchmarking the reconnect where we have multiple threads working concurrently, may not be too very desired. It would also come with an extra complexity of having to create separate Random instances with parametrized seeds for each of the threads. While certainly possible, I don't feel that this extra complexity is needed at this time.

So we can certainly do this in the future if we feel this would add a new dimension to the performance testing, but this doesn't seem necessary or justified currently.

Reply:
Entirely reasonable.  Definitely something to revisit in the future, as I have found many sources of performance degradation do not show up in steady-state or smooth situations, but become significant in the presence of sufficient variability.

Perhaps in the future we can create a _separate_ performance benchmark that runs over a much longer timeframe (so that delay can be close to real-world network delay, rather than extremely short) and tracks total "sleep" time as well as total "run" time so we can assess effective performance without noise from the simulated network delay.

Reply:
I filed https://github.com/hashgraph/hedera-services/issues/12343 to implement the delay fuzzing.

Reply:
FYI, the microsecond-level sleep is now implemented in the latest commit here. Also, there's now separate storage and network delays that can be tweaked independently.